### PR TITLE
chore: Fix clippy lint issues

### PR DESCRIPTION
**Pull request by guerinoni**

### CODE REVIEW

File: src/actions/delete_objects.rs [hunk 1]
```diff
@@ -69,6 +70,9 @@ impl<'a, I> DeleteObjects<'a, I>
 where
     I: Iterator<Item = &'a ObjectIdentifier>,
 {
+    /// Generate the XML body for the request.
+    /// # Panics
```
Comment (paolobarbolini) [inline src/actions/delete_objects.rs:74]: Looks better this way and markdown also prefers it.

```suggestion
    ///
    /// # Panics
    ///
```

File: src/base64.rs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 use ::base64::engine::general_purpose::STANDARD;
 use ::base64::engine::Engine;
 
-pub(crate) fn encode<T: AsRef<[u8]>>(input: T) -> String {
```
Comment (paolobarbolini) [inline src/base64.rs:4]: Wasn't it better with `pub(crate)`, making it more obvious that this is an internal only API?

File: src/signing/mod.rs [hunk 3]
```diff
@@ -69,31 +70,31 @@ where
...
     }
 
-    let a1;
-    let a2;
-    let standard_query = match token {
-        Some(token) => {
-            a1 = [
+    let mut a1 = [("", ""); 6];
+    let mut a2 = [("", ""); 5];
+    let standard_query = token.map_or_else(
```
Comment (paolobarbolini) [inline src/signing/mod.rs:75]: Given how big this piece is I'd prefer going with `let` syntax.

File: src/actions/delete_objects.rs (hunk 1 above)
Comment (guerinoni) [reply src/actions/delete_objects.rs:75]: Fixed

File: src/base64.rs (hunk 2 above)
Comment (guerinoni) [reply src/base64.rs]: Is inside a private module so it is warned about linters

**Pull request merged**